elide-hbase.md

```diff
@@ -0,0 +1,159 @@
+  private Set<Person> persons;
+
+  @OneToOne/@OneToMany/@ManyToOne/@ManyToMany
+  @DataStore(cls = HBaseStore.class)
+  @RowKeyFn(cls = RowKeyComputer.class)
+  public Set<Person> getPersons() { return Collections.emptySet(); }
+
+  static class RowKeyComputer implements Function<HobbyGroup, Object> { //could return a FilterScope maybe?
+    @Override
+    public Object apply(HobbyGroup hobbyGroup) {
+      byte[] rowKey = new byte[]{hobbyGroup.getId().byteValue()};
+      return rowKey;
+    }
+  }
+}
+```
+
+### Bridging DataStores
+1. Must use a `MultiplexDataStore`
+1. `PersistenceResource` will still be responsible for holding security check logics. `DataStoreTransaction` will have more to do with getting/setting attr, relations, etc. Methods in `DataStoreTransaction` aren't aware of securities at all
+1. Have a custom annotation `@DataStore(cls)`. This will be used to annotate relations of a different data store type in an entity bean.
+```
+@DataStore(cls=HBaseStore.class)
+public Collection<HBaseEntity> getHBaseEntities() { return Collections.emptySet(); }
+```
+1. Add a field `parentTransaction` to `DataStoreTransaction`, so that when using a `MultiplexTransaction`, a detail data store transaction object can get access to its parent(the MultiplexTransaction) and get to its peer(a transaction of a different data store type):
+```
+class DataStoreTransaction {
+  default DataStoreTransaction getParentTransaction() { return null; }
+}
+```
+1. Define `getRelation(Object entity, String relationName)` in `DataStoreTransaction`:
+```
+class DataStoreTransaction {
+  <T> Collection<T> getRelation(Object entity, String relationName) {
+    // Default implementation should call the getter method(just like the getValue() in PersistenceResource)
+    // if the getter method is annotated with @DataStore, retrieve the transaction for the specified datastore from the parent multiplex transaction,
+    // and then use it to loadObjects, with the Criteria/FilterScope computed by @RowKeyFn(or maybe rename it) function.
+  }
+}
+```
+1. Define `updateRelation(Object entity, String fieldName, Set<PersistentResource> resourceIdentifiers)` in `DataStoreTransaction`:
+```
+class DataStoreTransaction {
+  boolean setRelation(Object entity, String fieldName, Set<PersistentResource> resourceIdentifiers) {
+    // Default implementation should call the setter method(just like the setValue() in PersistenceResource)
+    // if the setter method is annotated with @DataStore, retrieve the transaction for the specified datastore from the parent multiplex transaction,
+    // and then use it to save(serialize), with the Criteria/FilterScope computed by @RowKeyFn(or maybe rename it) function.
+  }
+}
+```
+
+#### Getting & Setting attributes
+Attributes of an entity should be mostly within the same data store. It's not usual that a MySQL entity has an *attribute* that lives in HBase(in those cases, a better data model design would be making that attribute a relation.)
```

Comment:
While I agree this is probably unusual, I'm not sure we can make the case that this won't be needed.